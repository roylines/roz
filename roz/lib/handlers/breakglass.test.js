const {autoscaling} = require('../aws');
const {handle} = require('./breakglass.js');
const {send, pauseForEffect} = require('../telegram');

jest.mock('../aws');
jest.mock('../telegram');
jest.mock('lambda-log');

describe('breakglass handler', () => {
  beforeEach(() => jest.resetAllMocks());

  it('should set capacity and send notifications (ready first time)', async () => {
    autoscaling.get.mockResolvedValue({
      instance: {
        status: 'running',
      },
    });

    await handle();

    expect(autoscaling.set.mock.calls).toMatchSnapshot();
    expect(autoscaling.get.mock.calls).toMatchSnapshot();
    expect(send.mock.calls).toMatchSnapshot();
    expect(pauseForEffect.mock.calls).toMatchSnapshot();
  });
  it('should wait for ready', async () => {
    autoscaling.get.mockReset();
    autoscaling.get
      .mockResolvedValueOnce({
        instance: null,
      })
      .mockResolvedValueOnce({
        instance: {
          status: 'starting',
        },
      })
      .mockResolvedValueOnce({
        instance: {
          status: 'running',
        },
      });

    await handle();

    expect(autoscaling.set.mock.calls).toMatchSnapshot();
    expect(autoscaling.get.mock.calls).toMatchSnapshot();
    expect(send.mock.calls).toMatchSnapshot();
    expect(pauseForEffect.mock.calls).toMatchSnapshot();
  });
});
