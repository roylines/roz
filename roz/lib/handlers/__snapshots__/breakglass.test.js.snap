// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakglass handler should set capacity and send notifications (ready first time) 1`] = `
Array [
  Array [
    Object {
      "capacity": 1,
    },
  ],
]
`;

exports[`breakglass handler should set capacity and send notifications (ready first time) 2`] = `
Array [
  Array [],
]
`;

exports[`breakglass handler should set capacity and send notifications (ready first time) 3`] = `
Array [
  Array [
    Array [
      Array [
        "OK, I'm opening the bastion now.",
      ],
    ],
  ],
  Array [
    Array [
      Array [
        "Request made, waiting for it to be ready...",
      ],
    ],
  ],
  Array [
    Array [
      Array [
        "Right it's done! The bastion is open on undefined",
        "Don't do anything silly, I'm watching you...",
        "ðŸ‘€",
      ],
    ],
  ],
]
`;

exports[`breakglass handler should set capacity and send notifications (ready first time) 4`] = `Array []`;

exports[`breakglass handler should wait for ready 1`] = `
Array [
  Array [
    Object {
      "capacity": 1,
    },
  ],
]
`;

exports[`breakglass handler should wait for ready 2`] = `
Array [
  Array [],
  Array [],
  Array [],
]
`;

exports[`breakglass handler should wait for ready 3`] = `
Array [
  Array [
    Array [
      Array [
        "OK, I'm opening the bastion now.",
      ],
    ],
  ],
  Array [
    Array [
      Array [
        "Request made, waiting for it to be ready...",
      ],
    ],
  ],
  Array [
    Array [
      Array [
        "Still waiting...",
      ],
      Array [
        "One moment...",
      ],
      Array [
        "Tum-tee-tum...",
      ],
      Array [
        "Shouldn't be long now...",
      ],
      Array [
        "It's not quite ready...",
      ],
      Array [
        "Nearly there...",
      ],
      Array [
        "Only a couple of minutes more...",
      ],
    ],
  ],
  Array [
    Array [
      Array [
        "Still waiting...",
      ],
      Array [
        "One moment...",
      ],
      Array [
        "Tum-tee-tum...",
      ],
      Array [
        "Shouldn't be long now...",
      ],
      Array [
        "It's not quite ready...",
      ],
      Array [
        "Nearly there...",
      ],
      Array [
        "Only a couple of minutes more...",
      ],
    ],
  ],
  Array [
    Array [
      Array [
        "Right it's done! The bastion is open on undefined",
        "Don't do anything silly, I'm watching you...",
        "ðŸ‘€",
      ],
    ],
  ],
]
`;

exports[`breakglass handler should wait for ready 4`] = `
Array [
  Array [
    Object {
      "max": 10000,
      "min": 5000,
    },
  ],
  Array [
    Object {
      "max": 10000,
      "min": 5000,
    },
  ],
]
`;
